t/167-atomic-shdict.t

```diff
@@ -0,0 +1,658 @@
+# vim:set ft= ts=4 sw=4 et fdm=marker:
+use Test::Nginx::Socket::Lua;
+
+#worker_connections(1014);
+#master_process_enabled(1);
+#log_level('warn');
+
+#repeat_each(2);
+
+plan tests => repeat_each() * (blocks() * 3);
+
+#no_diff();
+no_long_string();
+#master_on();
+#workers(2);
+
+run_tests();
+
+__DATA__
+
+=== TEST 1: CAS int value
+--- http_config
+    lua_shared_dict dogs 1m;
+--- config
+    location = /test {
+        content_by_lua_block {
+            local dogs = ngx.shared.dogs
+            dogs:set("foo", 32)
+            local ok, err = dogs:cas("foo", 32, nil, 42)
+            local val = dogs:get("foo")
+            ngx.say(val, " ", type(val), " ", ok)
+        }
+    }
+--- request
+GET /test
+--- response_body
+42 number true
+--- no_error_log
+[error]
+
+
+
+=== TEST 2: CAS int value failed
+--- http_config
+    lua_shared_dict dogs 1m;
+--- config
+    location = /test {
+        content_by_lua_block {
+            local dogs = ngx.shared.dogs
+            dogs:set("foo", 32)
+            local ok, err = dogs:cas("foo", 31, nil, 42)
+            local val = dogs:get("foo")
+            ngx.say(val, " ", type(val), " ", ok, " ", err)
+        }
+    }
+--- request
+GET /test
+--- response_body
+32 number false false
+--- no_error_log
+[error]
+
+
```

Comment:
There are always only three blank lines between two use cases
maybe you should lint the code with `reindex` tool, https://github.com/openresty/openresty-devel-utils/blob/master/reindex